scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -662,9 +682,17 @@ def apply_filters(self, **kwargs):
                             if ts.harness_config.get('fixture') in h.fixtures:
                                 instance.run = True
 
+                if plat.simulation in [q['name'] for q in self.quarantine.simulation]:
```

Comment:
This method (`apply_filters`) is really big and not easy to read.
If there is separate Quarantine class, some methods could be added there (_search_comment, is_platform_under_quarantine, load_quarantinte_from_yaml etc.)